resourcepool/snapshot.go

```diff
@@ -0,0 +1,264 @@
+package resourcepool
```

Comment:
a lot of these functions seem very tied to `capacitycontroller`. why have them here? do you expect to reuse them in the scaler? it might be easier to keep them in `capacitycontroller` for now as we are still iterating on this controller quite a bit

Reply:
Yes, this code is shared with scaler. There was too much duplication. But these functions should be generic enough to use them in other contexts (soon in KubeScheduler).